src/20240528_Multi_GHOCrossChainLaunch/AaveV3Arbitrum_GHOCrossChainLaunch_20240528.sol

```diff
@@ -0,0 +1,230 @@
+import {UpgradeableTokenPool} from 'ccip/v0.8/ccip/pools/GHO/UpgradeableTokenPool.sol';
+import {RateLimiter} from 'ccip/v0.8/ccip/libraries/RateLimiter.sol';
+import {UpgradeableGhoToken} from 'gho-core/gho/UpgradeableGhoToken.sol';
+import {IGhoToken} from 'gho-core/gho/interfaces/IGhoToken.sol';
+
+/**
+ * @title GHO Cross-Chain Launch
+ * @author Aave Labs
+ * - Snapshot: https://snapshot.org/#/aave.eth/proposal/0x2a6ffbcff41a5ef98b7542f99b207af9c1e79e61f859d0a62f3bf52d3280877a
+ * - Discussion: https://governance.aave.com/t/arfc-gho-cross-chain-launch/17616
+ * @dev This payload consists of the following set of actions:
+ * 1. Deploy GHO
+ * 2. Deploy BurnMintTokenPool
+ * 3. Accept ownership of CCIP TokenPool
+ * 4. Configure CCIP TokenPool
+ * 5. Add CCIP TokenPool as GHO Facilitator
+ * 6. List GHO in Aave V3 Pool
+ * 7. Seed Aave Pool
+ */
+contract AaveV3Arbitrum_GHOCrossChainLaunch_20240528 is AaveV3PayloadArbitrum {
+  using SafeERC20 for IERC20;
+
+  address public immutable GHO;
+  address public immutable GHO_IMPL;
+  bytes32 public constant GHO_IMPL_DEPLOY_SALT = bytes32(uint256(keccak256('gho.token.impl')));
+  bytes32 public constant GHO_DEPLOY_SALT = bytes32(uint256(keccak256('gho.token')));
+  uint256 public constant GHO_SEED_AMOUNT = 1e18;
+  address public immutable CCIP_TOKEN_POOL;
+  address public immutable CCIP_TOKEN_POOL_IMPL;
+  bytes32 public constant CCIP_TOKEN_POOL_IMPL_DEPLOY_SALT =
+    bytes32(uint256(keccak256('gho.ccip.impl')));
+  bytes32 public constant CCIP_TOKEN_POOL_DEPLOY_SALT = bytes32(uint256(keccak256('gho.ccip')));
+  address public constant CCIP_ARM_PROXY = 0xC311a21e6fEf769344EB1515588B9d535662a145;
+  address public constant CCIP_ROUTER = 0x141fa059441E0ca23ce184B6A78bafD2A517DdE8;
+  uint256 public constant CCIP_BUCKET_CAPACITY = 1_000_000e18; // 1M
+  uint64 public constant CCIP_ETH_CHAIN_SELECTOR = 5009297550715157269;
+
+  constructor() {
+    // Predict GHO contract address
+    GHO_IMPL = ICreate3Factory(MiscArbitrum.CREATE_3_FACTORY).predictAddress(
+      GovernanceV3Arbitrum.EXECUTOR_LVL_1,
+      GHO_IMPL_DEPLOY_SALT
+    );
+    GHO = ICreate3Factory(MiscArbitrum.CREATE_3_FACTORY).predictAddress(
+      GovernanceV3Arbitrum.EXECUTOR_LVL_1,
+      GHO_DEPLOY_SALT
+    );
+    // Predict CCIP TokenPool contract address
+    CCIP_TOKEN_POOL_IMPL = ICreate3Factory(MiscArbitrum.CREATE_3_FACTORY).predictAddress(
+      GovernanceV3Arbitrum.EXECUTOR_LVL_1,
+      CCIP_TOKEN_POOL_IMPL_DEPLOY_SALT
+    );
+    CCIP_TOKEN_POOL = ICreate3Factory(MiscArbitrum.CREATE_3_FACTORY).predictAddress(
+      GovernanceV3Arbitrum.EXECUTOR_LVL_1,
+      CCIP_TOKEN_POOL_DEPLOY_SALT
+    );
+  }
+
+  /// @dev 6. List GHO in Aave V3 Pool
+  function newListings() public view override returns (IAaveV3ConfigEngine.Listing[] memory) {
+    IAaveV3ConfigEngine.Listing[] memory listings = new IAaveV3ConfigEngine.Listing[](1);
+
+    listings[0] = IAaveV3ConfigEngine.Listing({
```

Comment:
This params may need adjustments because GHO rate has changed since Chaos Labs analysis. Checking with them.

Reply:
Updated based on [last recs](https://governance.aave.com/t/arfc-gho-cross-chain-launch/17616/8?u=miguelmtz)